logger/xp.py

```diff
@@ -144,14 +145,25 @@ def log_config(self, config_dict, to_visdom=True):
         if to_visdom and self.use_visdom:
             self.plotter.plot_config(config_dict)
 
-    def log_with_tag(self, tag, idx=None):
+    def log_with_tag(self, pattern, idx=None, reset=False):
+        """ Log metrics from each tag matching the given pattern.
+        Pattern parameter must be in Unix shell-style wildcards format.
+        """
+        # log metrics of all tags matching the given pattern
+        for tag in fnmatch.filter(self.metrics, pattern):
+            self._log_with_tag(tag, idx, reset)
+
+    def _log_with_tag(self, tag, idx=None, reset=False):
 
         # log all metrics with given tag except Parents
         # (to avoid logging twice the information)
         for metric in self.metrics[tag].values():
             if isinstance(metric, ParentWrapper_):
                 continue
-            self.log_metric(metric, idx)
+            if reset:
```

Comment:
Let's pass the `reset` argument to the `log_metric` method (and modify accordingly the `log_metric` with an optional argument).  We can keep the `log_and_reset_metric` method as a legacy for now, but I like better this new approach you suggested.